pymc3/distributions/discrete.py

```diff
@@ -141,11 +147,14 @@ class Bernoulli(Discrete):
     p : float
         Probability of success (0 < p < 1).
     """
-    def __init__(self, p, *args, **kwargs):
-        super(Bernoulli, self).__init__(*args, **kwargs)
-        self.p = p
+    def __init__(self, p, ndim=None, size=None, dtype=None, *args, **kwargs):
+        self.p = T.as_tensor_variable(p)
```

Comment:
The `T.as_tensor_variable`?  If so, I'm of the opinion that everything internal to these fundamental PyMC3 objects should only deal with Theano (where possible).  There's little reason not to--from what I can tell--and it's a simpler assumption to hold than any possible mix of Theano/not-Theano objects.
Otherwise, do you mean the `__init__` signature?  I just put those keyword args in there by default.  In a few cases, we'll probably want to make use of them.
Actually, I've even been considering adding the old `shape` param for backward compatibility.